core/src/test/java/com/sequenceiq/cloudbreak/core/flow2/stack/update/userdata/UpdateUserDataFlowIntegrationTest.java

```diff
@@ -0,0 +1,249 @@
+import com.sequenceiq.cloudbreak.core.flow2.service.ReactorNotifier;
+import com.sequenceiq.cloudbreak.reactor.handler.userdata.UpdateUserDataHandler;
+import com.sequenceiq.cloudbreak.reactor.handler.userdata.UpdateUserDataOnProviderHandler;
+import com.sequenceiq.cloudbreak.domain.Resource;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.StackStatus;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.domain.view.ClusterView;
+import com.sequenceiq.cloudbreak.domain.view.StackView;
+import com.sequenceiq.cloudbreak.reactor.api.event.stack.userdata.UserDataUpdateRequest;
+import com.sequenceiq.cloudbreak.service.image.ImageService;
+import com.sequenceiq.cloudbreak.service.image.userdata.UserDataService;
+import com.sequenceiq.cloudbreak.service.resource.ResourceService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.util.StackUtil;
+import com.sequenceiq.cloudbreak.workspace.model.User;
+import com.sequenceiq.cloudbreak.workspace.model.Workspace;
+import com.sequenceiq.common.api.type.InstanceGroupType;
+import com.sequenceiq.common.api.type.Tunnel;
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+import com.sequenceiq.flow.core.FlowRegister;
+import com.sequenceiq.flow.domain.FlowLog;
+import com.sequenceiq.flow.repository.FlowLogRepository;
+
+@ActiveProfiles("integration-test")
+@ExtendWith(SpringExtension.class)
+class UpdateUserDataFlowIntegrationTest {
+
+    private static final String USER_CRN = "crn:cdp:iam:us-west-1:" + UUID.randomUUID() + ":user:" + UUID.randomUUID();
+
+    private static final String DATAHUB_CRN = "crn:cdp:iam:us-west-1:" + UUID.randomUUID() + ":cluster:" + UUID.randomUUID();
+
+    private static final long STACK_ID = 1L;
+
+    private static final int ALL_CALLED_ONCE = 2;
+
+    private static final int CALLED_ONCE_TILL_GENERATE_USERDATA = 1;
+
+    private static final String USER_DATA = "IS_CCM_V2_JUMPGATE_ENABLED=false";
+
+    @Inject
+    private FlowRegister flowRegister;
+
+    @Inject
+    private FlowLogRepository flowLogRepository;
+
+    @Inject
+    private ReactorNotifier reactorNotifier;
+
+    @MockBean(reset = MockReset.NONE)
+    private StackService stackService;
+
+    @MockBean
+    private UserDataService userDataService;
+
+    @MockBean
+    private ResourceService resourceService;
+
+    @MockBean
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @MockBean
+    private StackToCloudStackConverter cloudStackConverter;
+
+    @MockBean
+    private CredentialToCloudCredentialConverter credentialConverter;
+
+    @MockBean
+    private ImageService imageService;
+
+    @MockBean
+    private ResourceToCloudResourceConverter resourceToCloudResourceConverter;
+
+    @MockBean
+    private StackUtil stackUtil;
+
+    @Mock
+    private ResourceConnector<Object> resourcesApi;
+
+    private Stack stack;
```

Comment:
Should be converted to a local variable in `mockStack()`.

Reply:
ok